src/ussd_registration.js

```diff
@@ -0,0 +1,343 @@
+        App.call(self, 'state_start');
+        var $ = self.$;
+        var interrupt = true;
+
+        self.init = function() {
+
+            // Load self.contact
+            return self.im.contacts
+                .for_user()
+                .then(function(user_contact) {
+                   self.contact = user_contact;
+                });
+        };
+
+
+    // TEXT CONTENT
+
+        var questions = {
+            "state_timed_out":
+                "You have an incomplete registration. Would you like to continue with this registration?",
+            "state_auth_code":
+                "Welcome to Hello Mama! Please enter your unique personnel code. For example, 12345",
+            "state_msisdn":
+                "Please enter the mobile number of the person who will receive the weekly messages.  For example 0803304899",
+            "state_msg_receiver":
+                "Please select who will receive the messages on their phone:",
+            "state_pregnancy_status":
+                "Please select one of the following:",
+            "state_last_period_month":
+                "Please select the month the woman had her last period:",
+            "state_last_period_day":
+                "What day of the month did the woman start her last period? For example, 12.",
+            "state_baby_birth_date":
+                "Select the month & year the baby was born:",
+            "state_baby_birth_day":
+                "What day of the month was the baby born? For example, 12.",
+            "state_msg_language":
+                "Which language would this person like to receive these messages in?",
+            "state_msg_call_or_text":
+                "How would this person like to get messages?",
+            "state_receive_calls_days":
+                "We will call them twice a week. On what days would the person like to receive these calls?",
+            "state_receive_calls_time":
+                "Thank you. At what time would they like to receive these calls?",
+            "state_end_thank_you_calls":
+                "Thank you. The person will now start receiving calls on [day and day] between [time - time].",
+            "state_end_thank_you_texts":
+                "Thank you. The person will now start receiving messages three times a week."
+        };
+
+        var smss = {
+            "time_out":
+                "Please dial back in to *XXX*XX# to complete the Hello MAMA registration."
+        };
+
+        get_sms_text = function(msg_receiver) {
+            return msg_receiver === 'time_out'
+                ? smss.time_out : null;
+        };
+
+        var errors = {
+            "state_auth_code":
+                "Sorry, that is not a valid number. Please enter your unique personnel code. For example, 12345."
+        };
+
+        get_error_text = function(name) {
+            return errors[name] || "Sorry, that is not a valid number. " + questions[name];
```

Comment:
Change the string here to "Sorry, not a valid input. " so it's the same as FamilyConnect for now. We haven't received specific instructions for this project.